nano/core_test/difficulty.cpp

```diff
@@ -46,3 +64,24 @@ TEST (difficulty, network_constants)
 {
 	ASSERT_NEAR (16., nano::difficulty::to_multiplier (nano::network_constants::publish_full_threshold, nano::network_constants::publish_beta_threshold), 1e-10);
 }
+
+TEST (difficulty, overflow)
+{
+	// Overflow max
+	{
+		uint64_t base = std::numeric_limits<std::uint64_t>::max ();
+		uint64_t difficulty = std::numeric_limits<std::uint64_t>::max ();
+		double expected_multiplier = 1.001;
+
+		ASSERT_EQ (difficulty, nano::difficulty::from_multiplier (expected_multiplier, base));
```

Comment:
It would make sense to switch:
- difficulty -> expected_difficulty
- expected_multiplier -> multiplier

Same in the other test